<?xml version="1.0" ?>
<robot name="jetracer" xmlns:xacro="http://www.ros.org/wiki/xacro" >

<ros2_control name="GazeboSystem" type="system">
    <hardware>
        <plugin>gazebo_ros2_control/GazeboSystem</plugin>
    </hardware>
    <joint name="drive_back_right">
        <command_interface name="velocity">
            <param name="min">-10</param>
            <param name="max">10</param>
        </command_interface>
        <state_interface name="position" />
        <state_interface name="velocity" />
    </joint>
    <joint name="drive_back_left">
        <command_interface name="velocity">
            <param name="min">-10</param>
            <param name="max">10</param>
        </command_interface>
        <state_interface name="velocity" />
        <state_interface name="position" />
    </joint>
    <joint name="drive_front_right">
        <command_interface name="velocity">
            <param name="min">-10</param>
            <param name="max">10</param>
        </command_interface>
        <state_interface name="position" />
        <state_interface name="velocity" />
    </joint>
    <joint name="drive_front_left">
        <command_interface name="velocity">
            <param name="min">-10</param>
            <param name="max">10</param>
        </command_interface>
        <state_interface name="velocity" />
        <state_interface name="position" />
    </joint>
        <joint name="steering_left">
        <command_interface name="position">
            <param name="min">-0.174533</param>
            <param name="max">0.174533</param>
        </command_interface>
        <state_interface name="position" />
        <state_interface name="velocity" />
    </joint>
    <joint name="steering_right">
        <command_interface name="position">
            <param name="min">-0.174533</param>
            <param name="max">0.174533</param>
        </command_interface>
        <state_interface name="velocity" />
        <state_interface name="position" />
    </joint>
</ros2_control>

</robot>
